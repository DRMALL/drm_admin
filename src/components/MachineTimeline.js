

import React from 'react'
import Timeline from 'antd/lib/timeline'

export default props => {
  const { timelines } = props.state.machine
  console.log(timelines.length)
  return(
    <Timeline>
      {
        timelines.length&timelines.map((item, index) =>
          <Timeline.Item key={ index } >
            <span>{ item.time }</span>
            <span>{ item.timetype }</span>
            <span>{ item.timedes }</span>
          </Timeline.Item>
          )
      }
    </Timeline>
    )
}

// const MachineTimeline = ({ timelines }) => (
//   <div>
//     <Timeline>
//       {
//         timelines.length&timelines.map((item, index) =>
//           <Timeline.item key={ index } >
//             <span>{ item.time }</span>
//             <span>{ item.timetype }</span>
//             <span>{ item.timedes }</span>
//           </Timeline.item>
//           )
//       }
//     </Timeline>
//   </div>
//   )

// export default MachineTimeline
