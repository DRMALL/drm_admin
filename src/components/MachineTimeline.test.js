

import React from 'react'
import MachineTimeline from './MachineTimeline'
import renderer from 'react-test-renderer'


it('MachineTimeline test', () => {
  const timelines = [{line_type: '维修', line_des: '第一次维修' },
                     {line_type: '维修', line_des: '第一次维修' }]
  const tree = renderer.create(<MachineTimeline timelines={ timelines } />)
  expect(tree).toMatchSnapshot()
})
