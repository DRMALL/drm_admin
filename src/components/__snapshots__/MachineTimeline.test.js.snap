// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MachineTimeline test 1`] = `
<div>
  <ul
    className="ant-timeline machine-timeline"
  >
    <li
      className="ant-timeline-item"
    >
      <div
        className="ant-timeline-item-tail"
      />
      <div
        className="ant-timeline-item-head ant-timeline-item-head-blue"
        style={
          Object {
            "borderColor": null,
          }
        }
      />
      <div
        className="ant-timeline-item-content"
      >
        <div
          data-timeline-id={undefined}
          onBlur={undefined}
          onClick={[Function]}
          onFocus={undefined}
          onMouseDown={[Function]}
          onMouseEnter={undefined}
          onMouseLeave={undefined}
          onTouchStart={[Function]}
          style={
            Object {
              "width": 400,
            }
          }
        >
          <span
            className="machine-time-time"
          >
            2017-08-30
          </span>
          <span
            className="machine-time-type"
          >
            <b>
              维修
              :
            </b>
          </span>
          <span
            className="machine-time-des"
          >
            第一次维修
          </span>
        </div>
      </div>
    </li>
    <li
      className="ant-timeline-item ant-timeline-item-last"
    >
      <div
        className="ant-timeline-item-tail"
      />
      <div
        className="ant-timeline-item-head ant-timeline-item-head-blue"
        style={
          Object {
            "borderColor": null,
          }
        }
      />
      <div
        className="ant-timeline-item-content"
      >
        <div
          data-timeline-id={undefined}
          onBlur={undefined}
          onClick={[Function]}
          onFocus={undefined}
          onMouseDown={[Function]}
          onMouseEnter={undefined}
          onMouseLeave={undefined}
          onTouchStart={[Function]}
          style={
            Object {
              "width": 400,
            }
          }
        >
          <span
            className="machine-time-time"
          >
            2017-08-30
          </span>
          <span
            className="machine-time-type"
          >
            <b>
              维修
              :
            </b>
          </span>
          <span
            className="machine-time-des"
          >
            第一次维修
          </span>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
